x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorTemplateRegistryTests.java

```diff
@@ -0,0 +1,518 @@
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.oneOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class ConnectorTemplateRegistryTests extends ESTestCase {
+    private ConnectorTemplateRegistry registry;
+    private ThreadPool threadPool;
+    private VerifyingClient client;
+
+    @Before
+    public void createRegistryAndClient() {
+        threadPool = new TestThreadPool(this.getClass().getName());
+        client = new VerifyingClient(threadPool);
+        ClusterService clusterService = ClusterServiceUtils.createClusterService(threadPool);
+        registry = new ConnectorTemplateRegistry(clusterService, threadPool, client, NamedXContentRegistry.EMPTY);
+    }
+
+    @After
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        threadPool.shutdownNow();
+    }
+
+    public void testThatNonExistingComposableTemplatesAreAddedImmediately() throws Exception {
+        DiscoveryNode node = DiscoveryNodeUtils.create("node");
+        DiscoveryNodes nodes = DiscoveryNodes.builder().localNodeId("node").masterNodeId("node").add(node).build();
+        Map<String, Integer> existingComponentTemplates = Map.of(
+            ConnectorConstants.CONNECTOR_TEMPLATE_NAME + "-mappings",
+            ConnectorTemplateRegistry.REGISTRY_VERSION,
+            ConnectorConstants.CONNECTOR_TEMPLATE_NAME + "-settings",
+            ConnectorTemplateRegistry.REGISTRY_VERSION,
+            ConnectorConstants.CONNECTOR_SYNC_JOBS_TEMPLATE_NAME + "-mappings",
+            ConnectorTemplateRegistry.REGISTRY_VERSION,
+            ConnectorConstants.CONNECTOR_SYNC_JOBS_TEMPLATE_NAME + "-settings",
+            ConnectorTemplateRegistry.REGISTRY_VERSION,
+            ConnectorConstants.ACCESS_CONTROL_TEMPLATE_NAME,
+            ConnectorTemplateRegistry.REGISTRY_VERSION
+        );
+
+        ClusterChangedEvent event = createClusterChangedEvent(Collections.emptyMap(), existingComponentTemplates, nodes);
+
+        AtomicInteger calledTimes = new AtomicInteger(0);
+        client.setVerifier((action, request, listener) -> verifyComposableTemplateInstalled(calledTimes, action, request, listener));
+        registry.clusterChanged(event);
+        assertBusy(() -> assertThat(calledTimes.get(), equalTo(registry.getComposableTemplateConfigs().size())));
+
+        calledTimes.set(0);
+
+        // attempting to register the event multiple times as a race condition can yield this test flaky, namely:
+        // when calling registry.clusterChanged(newEvent) the templateCreationsInProgress state that the IndexTemplateRegistry maintains
+        // might've not yet been updated to reflect that the first template registration was complete, so a second template registration
+        // will not be issued anymore, leaving calledTimes to 0
```

Comment:
Just a nit, but I'd usually do a comment block with :
```
/*
this is
a multiline
java comment block
*/
```

Rather than:
```
// this reads 
// like three distinct
// inline comments
```

Reply:
Fixed in 91cbe54